src/jit/lower.cpp

```diff
@@ -831,26 +831,44 @@ GenTreePtr Lowering::NewPutArg(GenTreeCall* call, GenTreePtr arg, fgArgTabEntryP
     // Struct can be split into register(s) and stack on ARM
     if (info->isSplit)
     {
-        if (arg->OperGet() != GT_OBJ)
+        assert(arg->OperGet() != GT_OBJ || arg->OperGet() != GT_FIELD_LIST);
```

Comment:
This assert doesn't make sense. It will always be true. I think you meant to use `==` in these.